### ISSUE TITLE
Discrete and Continuous Actions in one Agent

### ISSUE THREAD

**Issue by rsfutch77** [Unity-Technologies/ml-agents#2101]
I am looking to train an agent to have both continuous and discrete actions. For example, the agent should be able to run at varying speeds in X and Z in any direction and decide whether or not to jump in Y. 

This is similar to the wall jump sample, but there the agent only decides whether it should rotate or move (not how much) so it can all be done with discrete actions. 

Will it be possible to have both types of actions in a brain in the future?
I'm guessing I don't want to make multiple brains (i.e. one for running and one for jumping)?
Can I accomplish this now with custom actions via custom protobuf?

**Comment by unityjeffrey** [comment]
hi @rsfutch77 you should be able to accomplish this using the custom actions via the custom protobuf by adding to CustomAction message.  See documentation here - https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Creating-Custom-Protobuf-Messages.md.

Do let us know if you have issues with it

**Comment by rsfutch77** [comment]
Thanks @unityjeffrey. I've found some issues while trying to do the Protobuf setup on windows. 

For reference, I've made some changes to the make_for_win.bat and the instructions on this branch:
https://github.com/rsfutch77/ml-agents/commits/protobuf_update

Unfortunately, after running the batch file and then `pip install $MLAGENTS_ROOT/ml-agents` as instructed in [Creating-Custom-Protobuf-Messages.md](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Creating-Custom-Protobuf-Messages.md), `mlagents-learn` fails to run. 

I've tried reverting all the .proto and .cs back to my original clone and running `pip install $MLAGENTS_ROOT/ml-agents` again, but `mlagents-learn` still fails with this result:

`Traceback (most recent call last):
  File "c:\users\rsfutch77\anaconda3\envs\ml-agents\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\rsfutch77\anaconda3\envs\ml-agents\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\rsfutch77\Anaconda3\envs\ml-agents\Scripts\mlagents-learn.exe\__main__.py", line 5, in <module>
  File "c:\users\rsfutch77\anaconda3\envs\ml-agents\lib\site-packages\mlagents\trainers\__init__.py", line 9, in <module>
    from .trainer_controller import *
  File "c:\users\rsfutch77\anaconda3\envs\ml-agents\lib\site-packages\mlagents\trainers\trainer_controller.py", line 19, in <module>
    from mlagents.trainers.ppo.trainer import PPOTrainer
ModuleNotFoundError: No module named 'mlagents.trainers.ppo'`

**Comment by unityjeffrey** [comment]
I believe you are correct @rsfutch77 - cc: @xiaomaogy, can you see the note above?

**Comment by xiaomaogy** [comment]
Yes @rsfutch77 you are correct. Can you issue a PR against develop branch to correct this?

**Comment by unityjeffrey** [comment]
Thank you for the discussion. We are closing this issue due to inactivity. Feel free to reopen it if you’d like to continue the discussion though.

**Comment by yakmyakmyakm** [comment]
Hi I've created a custom action. I can run mlagents-learn in the terminal and I see the unity logo like the documentation says. 

The issue is when I returned back to unity it looks like a bunch of references to the protobuf scripts which were generated do not match correctly.

<img width="622" alt="Screen Shot 2019-09-10 at 10 17 50 PM" src="https://user-images.githubusercontent.com/2358507/64670130-d4712800-d418-11e9-9542-3409e44f94ba.png">

I am on macOS Mojave
